tests/bot_tests/SOA_test.py

```diff
@@ -0,0 +1,169 @@
+"""unit tests for smart order accepter bot"""
+from gameplay_framework import GamePlay
+
+from baseline_bots.bots.smart_order_accepter_bot import SmartOrderAccepterBot
+from baseline_bots.bots.random_proposer_bot import RandomProposerBot
+from baseline_bots.bots.baseline_bot import BaselineBot, BaselineMsgRoundBot
+from baseline_bots.utils import (
+    MessagesData,
+    OrdersData,
+    get_state_value,
+    get_best_orders,
+    get_other_powers,
+    parse_orr_xdo,
+    parse_PRP,
+    dipnet_to_daide_parsing,
+    daide_to_dipnet_parsing
+)
+
+class TestSOABot():
+    def test(self):
+        self.test_play()
+        self.test_stance()
+        self.test_get_proposals()
+        self.test_get_best_orders()
+        self.test_gen_messages()
+
+    def test_play(self):
+        game_play = GamePlay(None, [RandomProposerBot, RandomProposerBot, SmartOrderAccepterBot], 3, True)
+        msgs, done = game_play.step()
+        print(msgs)
+        game_play = GamePlay(None, [RandomProposerBot, RandomProposerBot, SmartOrderAccepterBot], 3, True)
+        game_play.play()
+
+    def test_stance(self):
+        # score-based
+        SOA_bot = SmartOrderAccepterBot
+        bot_instances = [RandomProposerBot, RandomProposerBot, SOA_bot, RandomProposerBot ]
+        game_play = GamePlay(None, bot_instances, 3, True)
+        game_play.game.set_centers('AUSTRIA', ['VIE','TRI','BUD'], reset=True)
+        game_play.game.set_centers('ENGLAND', ['LON'])
+        game_play.game.set_centers('GERMANY', ['MUN', 'KIE', 'BER','BEL'])
+        game_play.game.set_centers(SOA_bot.power, ['PAR','BRE', 'MAR'])
```

Comment:
How is `SOA_bot.power` being used without an instantiated instance of `SOA_bot.power`?

Reply:
Right. that's silly mistake xD